src/Tribe/Attendee_Repository.php

```diff
@@ -663,4 +691,433 @@ protected function limit_list( $key, $list ) {
+
+		// Set the attendee arguments accordingly.
+		$this->set_attendee_args( $attendee_data );
+
+		// Update the attendee.
+		$saved = $this->save( $return_promise );
+
+		if ( $return_promise ) {
+			$repository = $this;
+
+			return $saved->then( static function() use ( $repository, $attendee_data ) {
+				// Trigger the update actions.
+				$repository->trigger_update_actions( $attendee_data );
+			} );
+		}
+
+		// Trigger the update actions.
+		$this->trigger_update_actions( $attendee_data );
+
+		return $saved;
+	}
+
+	/**
+	 * Set arguments for attendee.
+	 *
+	 * @since TBD
+	 *
+	 * @param array                         $attendee_data List of additional attendee data.
+	 * @param Tribe__Tickets__Ticket_Object $ticket        The ticket object or null if not relying on it.
+	 *
+	 * @throws Tribe__Repository__Usage_Error If the argument types are not set as expected.
+	 */
+	public function set_attendee_args( $attendee_data, $ticket = null ) {
+		$args = [
+			'attendee_id'       => null,
+			'title'             => null,
+			'full_name'         => null,
+			'email'             => null,
+			'ticket_id'         => $ticket ? $ticket->ID : null,
+			'post_id'           => $ticket ? $ticket->get_event_id() : null,
+			'order_id'          => null,
+			'order_attendee_id' => null,
+			'user_id'           => null,
+			'attendee_status'   => null,
+			'price_paid'        => null,
+			'optout'            => null,
+		];
+
+		$args = array_merge( $args, $attendee_data );
+
+		$attendee_id = null;
+
+		// Unset the attendee ID if found.
+		if ( isset( $args['attendee_id'] ) ) {
+			$attendee_id = $args['attendee_id'];
+
+			unset( $args['attendee_id'] );
+		}
+
+		// Do some extra set up if creating an attendee.
+		if ( null === $attendee_id ) {
+			// Default attendees to opted-out.
+			if ( null === $args['optout'] ) {
+				$args['optout'] = 1;
+			}
+
+			// If the title is empty, set the title from the full name.
+			if ( empty( $args['title'] ) && $args['full_name'] ) {
+				$args['title'] = $args['full_name'];
+
+				// Maybe add the Order ID.
+				if ( $args['order_id'] ) {
+					$args['title'] = $args['order_id'] . ' | ' . $args['title'];
+				}
+
+				// Maybe add the Order Attendee ID.
+				if ( null !== $args['order_attendee_id'] ) {
+					$args['title'] .= ' | ' . $args['order_attendee_id'];
+				}
+			}
+
+			// Maybe handle setting the User ID based on information we already have.
+			if ( empty( $args['user_id'] ) && ! empty( $args['email'] ) ) {
+				/**
+				 * Allow filtering whether to enable user lookups by Attendee Email.
+				 *
+				 * @since TBD
+				 *
+				 * @param boolean $lookup_user_from_email Whether to lookup the User using the Attendee Email if User ID is not set.
+				 * @param array   $args                   The arguments being set for this attendee.
+				 */
+				$lookup_user_from_email = (boolean) apply_filters( 'tribe_tickets_create_attendee_lookup_user_from_email', false, $args );
```

Comment:
:no_entry_sign: Short form type keywords must be used. Found: (boolean)

<sub>hash: 1cf0197174be7cf69472ad1ed93ab73a</sub>